x-pack/test/ingest_manager_api_integration/apis/epm/install_remove_assets.ts

```diff
@@ -324,6 +328,35 @@ export default function (providerContext: FtrProviderContext) {
         }
         expect(resSearch.response.data.statusCode).equal(404);
       });
+      it('should have removed the fields from the index patterns', async () => {
```

Comment:
I'm confused about which branch we expect to execute. The try or the catch?

Reply:
Both could happen. If a test case in another file calls `/setup` then the `system` and `endpoint` packages will be installed and will be present for the remainder of the tests (because they cannot be removed). If that is the case the `expect` in the `try` will work because the `logs-*` and `metrics-*` index patterns will still be present even after this test uninstalls its package.

If `/setup` was never called prior to this test, when the package is uninstalled the index pattern code checks to see if there are no packages installed and completely removes the `logs-*` and `metrics-*` index patterns. If that happens this code will throw an error and indicate that the index pattern being searched for was completely removed.

Reply:
Oh, interesting. Maybe we could add that as a comment?

Reply:
Yeah sure I'll put it as a comment.